src/hotspot/share/runtime/jniHandles.inline.hpp

```diff
@@ -55,7 +55,7 @@ inline oop JNIHandles::resolve_impl(jobject handle) {
   assert(!current_thread_in_native(), "must not be in native");
   oop result;
   if (is_jweak(handle)) {       // Unlikely
-    result = NativeAccess<ON_PHANTOM_OOP_REF|decorators>::oop_load(jweak_ptr(handle));
+    result = NativeAccess<ON_WEAK_OOP_REF|decorators>::oop_load(jweak_ptr(handle));
```

Comment:
Can you explain why? I believe I have pointed out the relevant sections in the specification, their changes, and also in the previous implementation. The current behavious is not only unintuitive, it also leads to the situation that it may lead to unintentional resurrection of finalized final or phantom references.

Reply:
I'm working on a longer reply.  I wanted to get a stake in the ground to block this change while I work on that.  It *is* a holiday here in the US.

Reply:
Sure, I am not in a rush. I guess I'm posting that mostly so that I can learn why we are in the situation that we're in, and see if it can be improved. Enjoy your thanksgiving!